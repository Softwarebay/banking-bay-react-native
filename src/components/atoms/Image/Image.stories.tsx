import { object } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StyleProp, View, ViewStyle } from 'react-native';
import { colors } from 'theme';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Image } from './Image';

const styleKnob = {
	width: '80%',
};
const darkViewStyle: StyleProp<ViewStyle> = {
	flex: 1,
	backgroundColor: colors.mainDark,
	justifyContent: 'center',
	alignItems: 'center',
};

storiesOf('Atom/Image', module)
	.addDecorator((getStory) => <ViewStory fullscreen>{getStory()}</ViewStory>)
	.add('Onboarding image 1', () => (
		<View style={darkViewStyle}>
			<Image name="onboarding1" style={object('style', styleKnob)} />
		</View>
	))
	.add('Onboarding image 2', () => (
		<View style={darkViewStyle}>
			<Image name="onboarding2" style={object('style', styleKnob)} />
		</View>
	))
	.add('Onboarding image 3', () => (
		<View style={darkViewStyle}>
			<Image name="onboarding3" style={object('style', styleKnob)} />
		</View>
	))
	.add('Card 1', () => (
		<View style={darkViewStyle}>
			<Image name="card1" style={object('style', styleKnob)} />
		</View>
	))
	.add('Card 2', () => (
		<View style={darkViewStyle}>
			<Image name="card2" style={object('style', styleKnob)} />
		</View>
	));
