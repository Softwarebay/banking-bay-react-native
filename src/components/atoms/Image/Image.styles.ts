import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
	card: {
		aspectRatio: 950 / 614,
	},
	img: {
		height: undefined,
	},
	onboarding: {
		aspectRatio: 843 / 1200,
	},
});
