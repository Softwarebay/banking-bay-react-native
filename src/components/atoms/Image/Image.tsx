import { ImageStyle, Image as RNImage, StyleProp } from 'react-native';

import {
	card1,
	card2,
	onboarding1,
	onboarding2,
	onboarding3,
	registrationIcon,
	successIcon,
} from 'assets/images';

import { styles } from './Image.styles';

export interface Props {
	name:
		| 'onboarding1'
		| 'onboarding2'
		| 'onboarding3'
		| 'card1'
		| 'card2'
		| 'success-icon'
		| 'registration-icon';
	style?: StyleProp<ImageStyle>;
}

// eslint-disable-next-line consistent-return
export const Image = ({ name, style }: Props) => {
	switch (name) {
		case 'onboarding1':
			return (
				<RNImage
					source={onboarding1}
					style={[styles.img, styles.onboarding, style]}
					testID="onboarding1-img-test-id"
				/>
			);
		case 'onboarding2':
			return (
				<RNImage
					source={onboarding2}
					style={[styles.img, styles.onboarding, style]}
					testID="onboarding2-img-test-id"
				/>
			);
		case 'onboarding3':
			return (
				<RNImage
					source={onboarding3}
					style={[styles.img, styles.onboarding, style]}
					testID="onboarding3-img-test-id"
				/>
			);
		case 'card1':
			return (
				<RNImage
					source={card1}
					style={[styles.img, styles.card, style]}
					testID="card1-img-test-id"
				/>
			);
		case 'card2':
			return (
				<RNImage
					source={card2}
					style={[styles.img, styles.card, style]}
					testID="card2-img-test-id"
				/>
			);
		case 'success-icon':
			return (
				<RNImage
					source={successIcon}
					style={[styles.img, styles.successIcon, style]}
					testID="success-icon-img-test-id"
				/>
			);
		case 'registration-icon':
			return (
				<RNImage
					source={registrationIcon}
					style={[styles.img, styles.registrationIcon, style]}
					testID="registration-icon-img-test-id"
				/>
			);
		// no default
	}
};
