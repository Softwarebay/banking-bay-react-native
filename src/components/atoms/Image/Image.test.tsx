import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Image } from './Image';

import type { Props } from './Image';

describe('Image', () => {
	const props: Omit<Props, 'name'> = {
		style: {
			width: 100,
		},
	};

	it('should match snapshot - onboarding1', () => {
		const tree = testRenderer
			.create(<Image {...props} name="onboarding1" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - onboarding2', () => {
		const tree = testRenderer
			.create(<Image {...props} name="onboarding2" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - onboarding3', () => {
		const tree = testRenderer
			.create(<Image {...props} name="onboarding3" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - card1', () => {
		const tree = testRenderer
			.create(<Image {...props} name="card1" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - card2', () => {
		const tree = testRenderer
			.create(<Image {...props} name="card2" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should match snapshot - success-icon', () => {
		const tree = testRenderer
			.create(<Image {...props} name="success-icon" />)
			.toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render onboarding1 image', () => {
		render(<Image {...props} name="onboarding1" />);

		const img = screen.getByTestId('onboarding1-img-test-id');

		expect(img).not.toBeNull();

		expect(img).toHaveStyle({
			width: 100,
		});
	});

	it('should render onboarding2 image', () => {
		render(<Image {...props} name="onboarding2" />);

		const img = screen.getByTestId('onboarding2-img-test-id');

		expect(img).not.toBeNull();

		expect(img).toHaveStyle({
			width: 100,
		});
	});

	it('should render onboarding3 image', () => {
		render(<Image {...props} name="onboarding3" />);

		const img = screen.getByTestId('onboarding3-img-test-id');

		expect(img).not.toBeNull();

		expect(img).toHaveStyle({
			width: 100,
		});
	});

	it('should render card1 image', () => {
		render(<Image {...props} name="card1" />);

		const img = screen.getByTestId('card1-img-test-id');

		expect(img).not.toBeNull();

		expect(img).toHaveStyle({
			width: 100,
		});
	});

	it('should render card2 image', () => {
		render(<Image {...props} name="card2" />);

		const img = screen.getByTestId('card2-img-test-id');

		expect(img).not.toBeNull();

		expect(img).toHaveStyle({
			width: 100,
		});
	});

	it('should render success-icon image', () => {
		render(<Image {...props} name="success-icon" />);

		const img = screen.getByTestId('success-icon-img-test-id');

		expect(img).not.toBeNull();

		expect(img).toHaveStyle({
			width: 100,
		});
	});
});
