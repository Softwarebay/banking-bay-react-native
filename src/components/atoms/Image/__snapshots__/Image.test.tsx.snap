// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should match snapshot - card1 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "height": undefined,
      },
      Object {
        "aspectRatio": 1.5472312703583062,
      },
      Object {
        "width": 100,
      },
    ]
  }
  testID="card1-img-test-id"
/>
`;

exports[`Image should match snapshot - card2 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "height": undefined,
      },
      Object {
        "aspectRatio": 1.5472312703583062,
      },
      Object {
        "width": 100,
      },
    ]
  }
  testID="card2-img-test-id"
/>
`;

exports[`Image should match snapshot - onboarding1 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "height": undefined,
      },
      Object {
        "aspectRatio": 0.7025,
      },
      Object {
        "width": 100,
      },
    ]
  }
  testID="onboarding1-img-test-id"
/>
`;

exports[`Image should match snapshot - onboarding2 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "height": undefined,
      },
      Object {
        "aspectRatio": 0.7025,
      },
      Object {
        "width": 100,
      },
    ]
  }
  testID="onboarding2-img-test-id"
/>
`;

exports[`Image should match snapshot - onboarding3 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "height": undefined,
      },
      Object {
        "aspectRatio": 0.7025,
      },
      Object {
        "width": 100,
      },
    ]
  }
  testID="onboarding3-img-test-id"
/>
`;

exports[`Image should match snapshot - success-icon 1`] = `
<Image
  source={1}
  style={
    Array [
      Object {
        "height": undefined,
      },
      Object {
        "aspectRatio": 1.0749464668094217,
      },
      Object {
        "width": 100,
      },
    ]
  }
  testID="success-icon-img-test-id"
/>
`;
